UI/org.eclipse.birt.report.designer.ui/src/org/eclipse/birt/report/designer/internal/ui/editors/schematic/editpolicies/TableXYLayoutEditPolicy.java

```diff
@@ -28,16 +31,27 @@
  *
  *
  */
-public class TableXYLayoutEditPolicy extends XYLayoutEditPolicy {
+public class TableXYLayoutEditPolicy extends ConstrainedLayoutEditPolicy {
+
+	private static final Dimension PREFERRED_SIZE = new Dimension(-1, -1);
 
 	/**
 	 * Constructor
 	 *
 	 * @param layout
 	 */
-	public TableXYLayoutEditPolicy(XYLayout layout) {
+	public TableXYLayoutEditPolicy(AbstractConstraintLayout layout) {
```

Comment:
If the layout is not used internally, does it make sense to keep it as a constructor argument?

I also wonder if it makes sense to keep the class name, when it no longer extends the XYLayoutEditPolicy.

Reply:
Good question, my first topic was to get it runable :o)
(Now we can talk about improvements.)

Reply:
Oh yea, you are completely right, I removed the parameter.